discretisedfield/field.py

```diff
@@ -2599,6 +2599,105 @@ def line(self, p1, p2, n=100):
+        of reduced dimensions intersects the axis at a point 'nearest' to the provided
+        value is returned. If the mesh is already 1 dimentional a numpy array of the
+        field values is returned.
+
+        If instead a tuple, list or a numpy array of length 2 is
+        passed as a value containing two real numbers (e.g. ``x=(1e-9, 7e-9)``), a sub
+        field is returned with minimum and maximum points along the selected axis,
+        'nearest' to the minimum and maximum of the selected values, respectively.
+
+        Parameters
+        ----------
+        args :
+
+            A string corresponding to the selection axis that belongs to
+            ``region.dims``.
+
+        kwarg :
+
+            A key corresponding to the selection axis that belongs to ``region.dims``.
+            The values are either a ``numbers.Real`` or list, tuple, numpy array of
+            length 2 containing ``numbers.Real`` which represents a point or a range of
+            points to be selected from the mesh.
+
+        Returns
+        -------
+        discretisedfield.Field or numpy.ndarray
+
+            An extracted field.
+
+        Examples
+        --------
+        1. Extracting the mesh at a specific point (``y=1``).
+
+        >>> import discretisedfield as df
+        ...
+        >>> p1 = (0, 0, 0)
+        >>> p2 = (5, 5, 5)
+        >>> cell = (1, 1, 1)
+        >>> mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+        >>> field = df.Field(mesh, nvdim=3, value=(1, 2, 3))
+        >>> plane_field = field.sel(y=1)
+        >>> plane_field.mesh.region.ndim
+        2
+        >>> plane_field.mesh.region.dims
+        ('x', 'z')
+        >>> plane_field.mean()
+        array([1., 2., 3.])
+
+        2. Extracting the xy-plane mesh at the mesh region center.
+
+        >>> plane_field = field.sel('z')
+        >>> plane_field.mesh.region.ndim
+        2
+        >>> plane_field.mesh.region.dims
+        ('x', 'y')
+        >>> plane_field.mean()
+        array([1., 2., 3.])
+
+        3. Specifying a range of points along axis ``x`` to be selected from mesh.
+
+        >>> selected_field = field.sel(x=(2, 4))
+        >>> selected_field.mesh.region.ndim
+        3
+        >>> selected_field.mesh.region.dims
+        ('x', 'y', 'z')
+
+        4. Extracting the mesh at a specific point on a 1D mesh.
+        >>> mesh = df.Mesh(p1=0, p2=5, cell=1)
+        >>> field = df.Field(mesh, nvdim=3, value=(1, 2, 3))
+        >>> field.sel('x')
+        array([1., 2., 3.])
+
+        """
+        dim, dim_index, _, sel_index = self.mesh._sel_convert_input(*args, **kwargs)
+
+        array = self.array[
+            dfu.assemble_index(
+                slice(None), self.mesh.region.ndim + 1, {dim_index: sel_index}
+            )
+        ]
+
+        try:
+            mesh = self.mesh.sel(*args, **kwargs)
+        except ValueError as e:
+            if "p1 and p2 must not be empty" not in str(e):
+                raise
```

Comment:
Does this require a `raise e` or simply `raise`? I have never used this, hence I do not know.

Reply:
This is on purpose `raise` without an exception and re-raises the currently handled exception.

`raise e` would technically also work but then the stack trace looks a bit different (without the `e` line 2694 does not appear in the stack trace).